sdk/scheduler/src/main/java/com/mesosphere/sdk/specification/DefaultServiceSpec.java

```diff
@@ -116,7 +119,8 @@ public int getApiPort() {
 
     @Override
     public String getZookeeperConnection() {
-        return zookeeperConnection;
+        // If no zookeeperConnection string is configured, fallback to the default value.
+        return StringUtils.isNotBlank(zookeeperConnection) ? zookeeperConnection : DEFAULT_ZK_CONNECTION;
```

Comment:
Maybe just use isBlank() and switch the cases? Could see it being slightly easier to parse